backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java

```diff
@@ -223,46 +225,66 @@ private Response filterInputOutput(Session session, Response response) {
 
   private Iterable<LineageEntryBatchResponse> filter(
       Session session, List<LineageEntryBatchResponse> lineageEntryBatchResponses) {
-    final Set<String> experimentRuns = new HashSet<>();
+    final Set<String> unfilteredExperimentRunIds = new HashSet<>();
+    for (LineageEntryBatchResponse lineageEntryBatchResponse : lineageEntryBatchResponses) {
+      List<LineageEntryBatchResponseSingle> lineageEntryBatchResponseItemsList =
+          lineageEntryBatchResponse.getItemsList();
+      for (LineageEntryBatchResponseSingle lineageEntryBatchResponseSingle :
+          lineageEntryBatchResponseItemsList) {
+        List<LineageEntry> itemList = lineageEntryBatchResponseSingle.getItemsList();
+        for (LineageEntry lineageEntry : itemList) {
+          if (lineageEntry.getDescriptionCase() == DescriptionCase.EXPERIMENT_RUN) {
+            unfilteredExperimentRunIds.add(lineageEntry.getExperimentRun());
+          }
+        }
+      }
+    }
+    final Set<String> filteredExperimentRunIds;
+    if (unfilteredExperimentRunIds.size() == 0) {
```

Comment:
As per intellij IDE suggestion, good to have 
```suggestion
    if (unfilteredExperimentRunIds.isEmpty()) {
```